### ISSUE TITLE
Thunar context menu option to move files will shred them if sent to standard Disposable

### ISSUE THREAD

**Issue by UndeadDevel** [QubesOS/qubes-issues#9657]
### Qubes OS release
4.2

### Brief summary
When sending files to another qube via the Thunar context menu ("Move to other qube") and selecting "Disposable VM ([...])" as the target, the new disp qube will be generated, will start, the files will be moved to it (so copied and then deleted in the origin qube) and the disp qube will immediately shut down, thus essentially deleting the files that were sent at the target location as well.
Since recovery doesn't seem too obvious, especially on SSDs with trim set to run after deletion, one could say that this actually shreds files (as opposed to the effect with "Copy to other qube", where at least only the copy will be shredded).

### Steps to reproduce
1. Select several files in an AppVM in Thunar
2. Right click and select "Move to other qube"
3. Start typing "Dispo" and choose one of the options that's not a template
4. Observe a new disposable starting up, the files disappearing and the disposable shutting down again

### Expected behavior
Either an error message or DispVM doesn't shut down (maybe launch a terminal or file browser when choosing "Move to other qube"?).

### Actual behavior
While the files technically are indeed moved, the effect is actually that they're deleted, which is not what that context menu option says. As this can lead to severe data loss I'm reporting as a bug and not enhancement request.

**Comment by ben-grande** [comment]
Yeah, this is hard, because the Qrexec Policy Prompt does not distinguish between RPCs, making it hard to filter fresh disposables from the list.

> Either an error message or DispVM doesn't shut down (maybe launch a terminal or file browser when choosing "Move to other qube"?).

This would require that every qube had a GUI.

**Comment by marmarek** [comment]
Should be easy to fix by adding `qubes.Filecopy * @anyvm @dispvm deny` rule into default policy.

**Comment by rustybird** [comment]
> Should be easy to fix by adding `qubes.Filecopy * @anyvm @dispvm deny` rule into default policy.

Unfortunately this doesn't work without a disposable template in the rule: https://github.com/QubesOS/qubes-issues/issues/10012#issuecomment-3263762681

**Comment by ben-grande** [comment]
> > I notice that adding a (high precedence) `qubes.Filecopy * @anyvm @dispvm deny` policy does not remove the "Disposable VM (...)" targets from the dialog for `qubes.Filecopy * @anyvm @anyvm ask`, whereas adding e.g. a `qubes.Filecopy * @anyvm personal deny` policy _does_ remove the "personal" target from this dialog.
> 
> Even better, either one of these works:
> 
>     qubes.Filecopy * @anyvm @dispvm:foo-dvm deny
>     qubes.Filecopy * @anyvm @dispvm:@tag:foo deny

Related:

- https://github.com/QubesOS/qubes-issues/issues/8227
- https://github.com/QubesOS/qubes-issues/issues/7723

> It's really just
> 
>     qubes.Filecopy * @anyvm @dispvm deny
> 
> that doesn't work. Maybe someone familiar with the [policy parser code](https://github.com/QubesOS/qubes-core-qrexec/tree/main/qrexec/policy) can spot the problem? 

Didn't read the parser code now, but from reading the rules, denying `@dispvm` would just deny the qube from trying to use the disposable template from the `default_dispvm` qube property. Which although it doesn't work, wouldn't be sufficient to hide all the other "Disposable VM (disposable-template-dvm)" entries, as there is a later rule allowing destination to be `@anyvm`. I think the parser is correct, it's just that there is no way to express in the policy rules that you want all qubes except fresh disposables.

Until we get more fields allowed in the policy, we might try something less intrusive, such as `@dispvm:@all`? Which would mean any fresh disposables of disposable templates. Should be only used for deny rules and never for allow. I didn't check if there is an order of rules that evaluating it works (see related issues above).
